src/utils/index.ts

```diff
@@ -617,6 +617,35 @@ export async function shellWithDocker(): Promise<vscode.Terminal> {
   return terminal;
 }
 
+interface WSServerRootFolderData {
+  redirectDotvscode: boolean;
+}
+
+const wsServerRootFolders = new Map<string, WSServerRootFolderData>();
+
+/**
+ * Add uri to the wsServerRootFolders map if eligible
+ */
+export async function addWsServerRootFolderData(uri: vscode.Uri): Promise<void> {
+  if (!schemas.includes(uri.scheme)) {
+    return;
+  }
+  const value: WSServerRootFolderData = {
+    redirectDotvscode: true,
+  };
+  if (isCSPFile(uri) && !["", "/"].includes(uri.path)) {
+    // A CSP-type root folder for a specific webapp that already has a .vscode/settings.json file must not redirect .vscode/* references
+    const api = new AtelierAPI(uri);
+    api
+      .headDoc(`${uri.path}/.vscode/settings.json`)
```

Comment:
Does this need to handle the case when `uri.path` ends with a `/`? Otherwise looks good.

Reply:
Fixed